### PR TITLE
Selenium tests

### PR DESCRIPTION
**Pull request by dnaranjo89**

### CODE REVIEW

File: tests/selenium/browserstack_base.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import os
+from os.path import join
+import django
+from django.test import LiveServerTestCase
+from selenium import webdriver
+from echaloasuerte.settings.common import SITE_ROOT
+from server.mongodb.driver import MongoDriver
+
+LOCAL_TESTS = os.environ.get('RUN_TESTS_LOCALLY')
```
Comment (Mariocj89) [inline tests/selenium/browserstack_base.py]: what about having a env var for the config of browser stack and if it is not present you just run them locally? No file, nothing else

File: .gitignore [hunk 2]
```diff
@@ -63,3 +63,6 @@ docs/_build/
 
 # Secret Key file
 secret.txt
+
+# Secret Key Browserstack
+secret_tests.txt
```
Comment (Mariocj89) [inline .gitignore]: ??

File: requirements.txt [hunk 3]
```diff
@@ -5,3 +5,4 @@ pymongo==2.7.2
 pytz==2014.10
 django-tastypie==0.12.2
 django-tastypie-swagger==0.1.4
+selenium==2.47.1
```
Comment (Mariocj89) [inline requirements.txt]: ?? not here :)

Comment (Mariocj89) [inline requirements.txt]: add it to travis.yml

File: tests/selenium/browserstack_base.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                           'os': 'OS X',
+                           'os_version': 'Yosemite',
+                           'resolution': '1024x768',
+                           'browserstack.local': True,
+                           'browserstack.debug': True
+                           }
+            self.driver = webdriver.Remote(command_executor=test_url, desired_capabilities=desired_cap)
+        else:
+            # PhantomJS (Silent mode)
+            self.driver = webdriver.PhantomJS('C:/phantomjs/bin/phantomjs.exe')
```
Comment (Mariocj89) [inline tests/selenium/browserstack_base.py]: no :)

File: tests/selenium/test_draw.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+class NormalDrawTest(BrowserStackTest):
+    """
+    Test that all draw types can be accessed and generate results.
+    """
+
+    def setUp(self):
+        super(NormalDrawTest, self).setUp()
+
+        # Load starting page
+        self.driver.get(self.base_url + "/")
```
Comment (Mariocj89) [inline tests/selenium/test_draw.py]: why dont adding the / to the base url

File: tests/selenium/browserstack_base.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+class BrowserStackTest(LiveServerTestCase):
+
+    def __init__(self, *args, **kwargs):
+        super(BrowserStackTest, self).__init__(*args, **kwargs)
+        django.setup()
+
+    def wait(self, seconds=10):
+        # Set up a default implicit wait for 10 seconds
+        self.driver.implicitly_wait(seconds)
```
Comment (Mariocj89) [inline tests/selenium/browserstack_base.py]: "Once set, the implicit wait is set for the life of the WebDriver object instance."

File: tests/selenium/user_test.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        result = driver.find_elements_by_id('account-dropdown')
+        self.assertNotEqual([], result)
+
+    def test_login_in_screen(self):
+        """ User login (from login screen) """
+        driver = self.driver
+        self.login()
+        self.wait()
+        result = driver.find_elements_by_id('account-dropdown')
+        self.assertNotEqual([], result)
```
Comment (Mariocj89) [inline tests/selenium/user_test.py]: assertTrue

File: tests/selenium/test_draw.py (hunk 5 above)
Comment (dnaranjo89) [inline tests/selenium/test_draw.py]: it actually wouldn't be necessary. It simply more semantic, to know that we are starting in the root directory.

File: tests/selenium/user_test.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    Test that all draw types can be accessed and generate results.
+    """
+
+    def remove_user(self, user_id):
+        self.db._users.remove({'_id': user_id})
+
+    def setUp(self):
+        super(UserTest, self).setUp()
+
+        # delete user if exists
```
Comment (Mariocj89) [inline tests/selenium/user_test.py]: bad comment

File: tests/selenium/browserstack_base.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+                           'os': 'OS X',
+                           'os_version': 'Yosemite',
+                           'resolution': '1024x768',
+                           'browserstack.local': True,
+                           'browserstack.debug': True
+                           }
+            self.driver = webdriver.Remote(command_executor=test_url, desired_capabilities=desired_cap)
+        else:
+            # PhantomJS (Silent mode)
+            self.driver = webdriver.PhantomJS('phantomjs.exe')
```
Comment (Mariocj89) [inline tests/selenium/browserstack_base.py]: do you need to pass "phantomjs.exe" if you have it in the path????

File: tests/selenium/browserstack_base.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+            # Firefox (Graphic mode)
+            # self.driver = webdriver.Firefox()
+
+            self.driver.set_window_size(1000, 900)
+
+        # Set up implicit wait for 10 seconds
```
Comment (Mariocj89) [inline tests/selenium/browserstack_base.py]: remove unuseful comment

File: tests/selenium/test_draw.py [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+    def test_dice(self):
+        """Selenium: Dice draw (normal draw)"""
+        driver = self.driver
+        draw_box = driver.find_element_by_id("dice-draw")
+        draw_box.click()
+        toss_btn = driver.find_element_by_id("toss")
+        toss_btn.click()
+        result = driver.find_elements_by_id("results")
+        self.assertNotEqual([], result)
```
Comment (Mariocj89) [inline tests/selenium/test_draw.py]: be consistent, in the next you do assert True

File: tests/selenium/browserstack_base.py (hunk 9 above)
Comment (dnaranjo89) [inline tests/selenium/browserstack_base.py]: I'll check it and remove if it not necessary

**Pull request closed without merging**